src/chapters/typescript/functions.rst

```diff
@@ -0,0 +1,139 @@
+Functions in TypeScript
+=======================
+
+When creating functions in TypeScript, we have many of the same options as in JavaScript.
+We can make anonymous functions, give them a different number of parameters, and so on.
+However, when working with functions in TypeScript, we have to keep *data types* in mind.
+
+Declaring Functions
+-------------------
+
+Named Functions
+^^^^^^^^^^^^^^^
+
+Let's take a look at a TypeScript function declaration.
+
+.. admonition:: Example
+
+   We want to declare a function called ``myFunction``.
+   ``myFunction`` has one parameter, a number called ``x``.
+   ``myFunction`` returns the value of ``x`` multiplied by 2.
+
+   With this in mind, the declaration in TypeScript would look like:
+
+   .. sourcecode:: typescript
+
+      function myFunction(x: number): number {
+         return x*2;
+      }
+
+   Here you can see that we provided the type of the parameter and the type of the value that is returned after the colons.
+   If ``x`` was a string or a boolean, then we would replace ``number`` with the data type of ``x``.
+
+What if the function doesn't return a value? In these cases, we use ``void`` as the return type.
+
+.. admonition:: Example
+
+   Let's change up ``myFunction`` a little bit!
+   ``myFunction`` still has one parameter, our number called ``x``, however, it doesn't return a specific value.
+
+   .. sourcecode:: typescript
+
+      let y: number = 0;
+
+      function myFunction(x: number): void {
+         y = x*2;
+      }
+
+   Instead of returning the value of ``x`` multiplied by 2, ``myFunction`` now assigns the value of ``x`` multiplied by 2 to another variable, ``y``.
+   We can now use ``void`` to specify that no value is returned.
+
+Anonymous Functions
+^^^^^^^^^^^^^^^^^^^
+
+For an anonymous function, you still need to provide types for the value returned and the parameters.
+
+.. admonition:: Example
+
+   Now we want to declare an anonymous function, ``myFunction``, which has one parameter, a number called ``x``, and returns the value of ``x`` multiplied by 2.
+
+   .. sourcecode:: typescript
+
+      let myFunction = function(x: number): number { return x*2; };
+
+Just as we did above with the named function, we need to make sure that we include the data type for the parameters and the return type of the function.
+
+Optional Parameter
+------------------
+
+In JavaScript, you can declare a function with 5 parameters and only give it 2 when it comes time to use it.
+In TypeScript, that is not the case.
+In TypeScript when you declare a function with 5 parameters, you have to give the function 5 arguments when calling it, unless you make some of those parameters optional.
+As with JavaScript functions, optional parameters can be omitted.
+This means that when you are calling the function, you can leave off the optional parameter(s).
```

Comment:
I think lines 72 and 73 say the same thing, no? i would keep one and lose the other.

Reply:
Done! Thanks!